cpp/src/parquet/properties.h

```diff
@@ -56,10 +60,32 @@ class PARQUET_EXPORT ReaderProperties {
 
   bool is_buffered_stream_enabled() const { return buffered_stream_enabled_; }
 
+  bool is_coalesced_stream_enabled() const { return coalesced_stream_enabled_; }
```

Comment:
So rather than having this flag here, what if you put a prebuffer option in `ArrowReadProperties`? Then `FileReader::PreBuffer` will always prebuffer if it's called